compiler/noirc_evaluator/src/ssa/opt/assert_constant.rs

```diff
@@ -108,27 +113,71 @@ fn evaluate_static_assert(
     instruction: InstructionId,
     arguments: &[ValueId],
 ) -> Result<bool, RuntimeError> {
-    if arguments.len() != 2 {
+    if arguments.len() < 2 {
         panic!("ICE: static_assert called with wrong number of arguments")
     }
 
-    if !function.dfg.is_constant(arguments[1]) {
-        let call_stack = function.dfg.get_instruction_call_stack(instruction);
-        return Err(RuntimeError::StaticAssertDynamicMessage { call_stack });
+    // All of the arguments representing the message must be constants
+    for arg in arguments.iter().skip(1) {
+        if !function.dfg.is_constant(*arg) {
+            let call_stack = function.dfg.get_instruction_call_stack(instruction);
+            return Err(RuntimeError::StaticAssertDynamicMessage { call_stack });
+        }
     }
 
     if function.dfg.is_constant_true(arguments[0]) {
-        Ok(false)
-    } else {
-        let call_stack = function.dfg.get_instruction_call_stack(instruction);
-        if function.dfg.is_constant(arguments[0]) {
-            let message = function
-                .dfg
-                .get_string(arguments[1])
-                .expect("Expected second argument to be a string");
+        return Ok(false);
+    }
+
+    let call_stack = function.dfg.get_instruction_call_stack(instruction);
+    if !function.dfg.is_constant(arguments[0]) {
+        return Err(RuntimeError::StaticAssertDynamicPredicate { call_stack });
```

Comment:
Can this error be delayed until we've attempted to calculate the message?

Reply:
Ah,I actually made that intentional do we don't have to calculate the message if it's not needed. Why you'd prefer it the other way around?

Reply:
Yes: I think it would be easier to debug with the message